### DISCUSSION TITLE
Build examples with a custom `planetiler-core`

### DISCUSSION THREAD

**Discussion by wipfli** [onthegomap/planetiler#529] [Q&A] [answered]
I would like to build the `planetiler-examples` with a custom `planetiler-core`. At the moment, I think the examples download the core from maven. How could I reference the code in the `planetiler-core` project from the `planetiler-examples`?

My goal is to play around with the buffer - union - unbuffer behavior in `mergeNearbyPolygons`. In particular, I would like to see how the visual output would look like if we only did buffer - union without subsequent unbuffering...

**Comment by bdon** [comment, marked as answer]
Here's how I do it:

Remove the cached maven artifact: 

```
rm -rf ~/.m2/repository/com/onthegomap/
```

in the `planetiler-core` directory, install your compiled custom planetiler-core into the local maven repository: 

```
mvn install -Pflatten -DskipTests
```

When you're building an example, use the `-o` (offline) flag with maven so it doesn't check for a new artifact from maven central

Any other alternatives?

**Reply by wipfli** [reply]
Thanks the tip, @bdon. This looks reasonable but I get an error:

```
git clone git@github.com:onthegomap/planetiler.git
cd planetiler/
git submodule update --init --recursive  
cd planetiler-core
../mvnw clean package
rm -rf ~/.m2/repository/com/onthegomap/
../mvnw install -Pflatten -DskipTests
cd ../planetiler-examples
../mvnw clean package --file standalone.pom.xml -o
```

produces the following error:

```
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------< com.onthegomap.planetiler:planetiler-examples >------------
[INFO] Building Example Planetiler Project HEAD
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.178 s
[INFO] Finished at: 2023-03-23T17:56:53+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project planetiler-examples: Could not resolve dependencies for project com.onthegomap.planetiler:planetiler-examples:jar:HEAD: Failed to collect dependencies at com.onthegomap.planetiler:planetiler-core:jar:0.6-SNAPSHOT: Failed to read artifact descriptor for com.onthegomap.planetiler:planetiler-core:jar:0.6-SNAPSHOT: Cannot access nexus-snapshots (https://s01.oss.sonatype.org/content/repositories/snapshots/) in offline mode and the artifact com.onthegomap.planetiler:planetiler-parent:pom:0.6-SNAPSHOT has not been downloaded from it before. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

**Reply by wipfli** [reply]
If I only do `rm -rf ~/.m2/repository/com/onthegomap/planetiler/planetiler-core` and keep the `planetiler-parent` folder, it compiles.

**Reply by wipfli** [reply]
Yay, it worked:

![image](https://user-images.githubusercontent.com/53421382/227287808-49c4388e-56c5-4692-b0cd-1e3da87d44e9.png)

**Reply by wipfli** [reply]
Thanks @bdon

**Reply by msbarry** [reply]
planetiler-examples also get linked into the parent pom, so you can just change both, then run `scripts/build.sh` and run `java -cp planetiler-dist/target/planetiler-dist-0.6-SNAPSHOT-with-deps.jar com.onthegomap.planetiler.examples.BikeRouteOverlay`

**Reply by wipfli** [reply]
That worked, thanks @msbarry
